src/scales/categoryScale.ts

```diff
@@ -130,8 +154,34 @@ namespace Plottable.Scales {
     }
 
     public scale(value: string): number {
-      // scale it to the middle
-      return this._d3Scale(value) + this.rangeBand() / 2;
+      // Determine the middle of the range band for the value
+      let untransformed = this._d3Scale(value) + this._d3Scale.rangeBand() / 2;
+      // Convert to screen space
+      return this._d3TransformationScale(untransformed);
+    }
+
+    public magnify(magnifyAmount: number, centerValue: number) {
```

Comment:
Is this the "single source of truth" for panning/zooming a scale? If so, PanZoomInteraction should use these